logstash-core/src/main/java/org/logstash/execution/DiscoverPlugins.java

```diff
@@ -0,0 +1,32 @@
+package org.logstash.execution;
+
+import java.lang.reflect.Constructor;
+import java.util.Set;
+import org.reflections.Reflections;
+
+/**
+ * Quick demo of plugin discovery showing that the solution wouldn't require anything beyond
+ * the plugin classes on the classpath.
+ */
+public final class DiscoverPlugins {
+
+    public static void main(final String... args) throws NoSuchMethodException {
```

Comment:
What are your thoughts on isolated class loaders per plugin ?

Reply:
I agree with your assessment of introduction of alot of complexity. However, if using a security managed LS is in the near future, it may be worth tackling that complexity sooner then later since its usually easier to bake things like that in rather then layer them in them later.  Also if security managed LS is in the future, then maybe Java9 modules provides some wins there without isolated class loading (just speculation, I haven't researched).

Reply:
I don't think a security managed LS is that close. I'd rather bite off that complexity a bit later.